### PR TITLE
Adding types to declaration merge for route transact option in typesc…

### LINKED ISSUE #141: Typescript types missing for FastifyRequest and RouteShorthandOptions when using the transact route option
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.0.0

### Plugin version

5.0.0

### Node.js version

16.16.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.4

### Description

When attempting to use the transact route option using typescript, passing in the transact route options and referencing `pg` from the request object throws typescript type errors. This is because because `pg` has not been added to FastifyRequest and 

```
interface FastifyPostgresRouteOptions {
  transact: boolean | string;
}
```
has not been added for `pg` for RouteShorthandOptions

### Steps to Reproduce

```
fastify.post(
    '/',
    {
      pg: { transact: true }, // Throws type error for RouteShorthandOptions
    },
    async (req, _reply) => {
      req.pg.query('SELECT * FROM table') // Throws type error for FastifyRequest
    }
  );
```

### Expected Behavior

Using the route transact option should not cause compile errors in typescript

### PR DESCRIPTION
**Pull request by bdun94**
Extended `FastifyRequest` and `RouteShorthandOptions` so that the transact route option is usable in typescript. Basically added in the missing definitions in the typescript declaration merging so that when using the transact route typescript compile errors are not thrown

resolves #141 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included 
- [x] documentation is changed or added (don't think it's necessary for this change?)
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — CHANGES_REQUESTED** [review]
Can you clean up the unnecessary styles changes?
e.g. quotes

**Comment by bdun94** [comment]
sorry about that @climba03003 should've caught that, let me know if there's anything else that needs to be changed

**Review by climba03003 — APPROVED** [review]

File: index.d.ts [hunk 1]
```diff
@@ -41,6 +45,13 @@ declare module 'fastify' {
   export interface FastifyInstance {
     pg: PostgresDb & Record<string, PostgresDb>;
   }
+
+  export interface FastifyRequest {
+    pg?: Pg.PoolClient;
+  }
+  export interface RouteShorthandOptions {
```
Comment (Uzlopak) [inline index.d.ts:52]: nitpick:

```suggestion

  export interface RouteShorthandOptions {
```

**Review by Uzlopak — APPROVED** [review]
LGTM, just a nitpick

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**